src/commonMain/kotlin/app/cash/turbine/flow.kt

```diff
@@ -193,15 +202,20 @@ public fun <T> Flow<T>.testIn(
   return turbine
 }
 
-private fun <T> Flow<T>.collectTurbineIn(scope: CoroutineScope, timeout: Duration?, name: String?): ReceiveTurbine<T> {
+private fun <T> Flow<T>.collectTurbineIn(
+  scope: CoroutineScope,
+  timeout: Duration?,
+  name: String?,
+  collectContext: CoroutineContext,
+): ReceiveTurbine<T> {
   // Use test-specific unconfined if test scheduler is in use to inherit its virtual time.
   @OptIn(ExperimentalCoroutinesApi::class) // UnconfinedTestDispatcher is still experimental.
   val unconfined = scope.coroutineContext[TestCoroutineScheduler]
     ?.let(::UnconfinedTestDispatcher)
     ?: Unconfined
 
   val output = Channel<T>(UNLIMITED)
-  val job = scope.launch(unconfined, start = UNDISPATCHED) {
+  val job = scope.launch(collectContext + unconfined, start = UNDISPATCHED) {
```

Comment:
I had thought that overriding the dispatcher here would be a big reason folks would want to use this. I do think it would be surprising if I tried to use that parameter for that purpose and found it didn't work.

```suggestion
  val collectDispatcher = collectContext[CoroutineDispatcher] ?: unconfined
  val job = scope.launch(collectContext + collectDispatcher, start = UNDISPATCHED) {
```

Reply:
We should throw if you try to do that. There's no reason to change the dispatcher on which we write into the channel, and anything other than unconfined would open up the potential for correctness problems since you make things like `expectMostRecentItem()` (or whatever it's called) racy.

Reply:
> There's no reason to change the dispatcher on which we write into the channel

If this is so (and I think I agree, even though I hate unconfined - we don't do anything inside `collect`, apart from call `trySend`, so any dispatcher hop on `emit` can only be bad), then maybe the right change is this:

```kotlin
flowOn(collectContext).collect { output.trySend(it) }
```

Reply:
Yes, that seems nice. Should the parameter be called `flowContext`? `flowOnContext`? Something else?

Reply:
I'm pretty sure this whole feature isn't needed. Just call `flowOn`!